docs/reference/ccr/upgrading.asciidoc

```diff
@@ -0,0 +1,49 @@
+[role="xpack"]
+[testenv="platinum"]
+[[ccr-upgrading]]
+== Upgrading clusters
+
+In case of upgrading clusters, clusters that are actively using CCR, require a
+slightly different upgrade approach. Otherwise index following may fail during
+a rolling upgrade, because of the following reasons:
+
+* If a new index setting or mapping type is replicated from an upgraded cluster
+  to a non-upgraded cluster then the non-upgraded cluster will reject that and
+  will fail index following.
+* Lucene is not forwards compatible and when index following is falling back to
+  file based recovery then a node in a non-upgraded cluster will reject index
+  files from a newer Lucene version compared to what it is using.
+
+Rolling upgrading clusters with CCR is different in case of uni-directional
+index following and bi-directional index following.
+
+[float]
+=== Uni-directional index following
+
+In a uni-directional setup between two clusters, one cluster contains only
+leader indices, and the other cluster contains only follower indices following
+indices in the first cluster.
+
+In this setup, the cluster with follower indices should be upgraded
+first and the cluster with leader indices should be upgraded last.
+If clusters are upgraded in this order then index following can continue
+during the upgrade without downtime.
+
+Note that a chain index following setup can also be upgraded in this way.
+For example if there is a first cluster that contains all leader indices,
+a second cluster that follows indices in the first cluster and a third
+cluster that follows indices in the second clusters. In this case the
+third cluster should be upgraded first, then the second cluster and
```

Comment:
Maybe we should avoid referring to the clusters as "first", "second", and "third" only to turn around and say "third cluster [...] first", that's confusing. How about `A`, `B`, and `C`?

Reply:
pushed: a91bbda212cd7d28cd3b86526ad87703633ef473